nixos/modules/services/databases/cockroachdb.nix

```diff
@@ -0,0 +1,162 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+
+  cfg = config.services.cockroachdb;
+
+  cockroachdb = cfg.package;
+
+in
+
+{
+
+  ###### interface
+
+  options = {
+
+    services.cockroachdb = {
+
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = "
+          Whether to enable the CockroachDB server.
+        ";
+      };
+
+      package = mkOption {
+        type = types.package;
+        example = literalExample "pkgs.cockroachdb";
+        description = "
+          Which CockroachDB derivation to use.
+        ";
+      };
+
+      cache = mkOption {
+        type = types.str;
+        default = "25%";
+        description = "The total size for caches. This can be a percentage or any bytes-based unit.";
+      };
+
+      certsDir = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        description = "The path to the certificate directory.";
+      };
+
+      httpHost = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = "The host to bind to for Admin UI HTTP requests.";
+      };
+
+      httpPort = mkOption {
+        type = types.int;
+        default = 8080;
+        description = "The port to bind to for Admin UI HTTP requests.";
+      };
+
+      host = mkOption {
+        type = types.nullOr types.str;
+        default = "localhost";
+        description = "The hostname or IP address to listen on for intra-cluster and client communication.";
+      };
+
+      insecure = mkOption {
```

Comment:
We try to make insecure mode an explicit choice instead of a default wherever possible (unfortunately automating setup of secure clusters is still kind of an open problem).